### DISCUSSION TITLE
Linux ver. of Ruffle shows as "shared library" not executable and doesn't open for gold. WHY is no one talking about this?

### DISCUSSION THREAD

**Discussion by CurHob** [ruffle-rs/ruffle#15914] [Q&A] [answered]
Preface: I am using Ubuntu, and just updated to the latest version, if it helps.

Also, the reason why I try to specifically use the Linux standalone version is that I want to TAS a game in Flash, and the documentation on TASvideos says libTAS requires the application assigned to it is specifically ruffle as a standalone. So I need to run that to TAS Flash files using Ruffle. Not to mention the game I am trying to use doesn't work in browser, perhaps because it is a series of SWF files that each open as it plays.

Trying to install libTAS was annoying for me because I tried to follow instructions involving a whole bunch of sudo apt install commands in Terminal, which may be part of why this isn't working for me. (although I doubt that because like EVERY OTHER PROGRAM shows up as .deb!). I had since discovered that it's much easier to install the .deb version of it and get that working. So then all that's left is to get Ruffle and assign it to the program, right?

Sigh... Of course not. When I got it from the Other Downloads page for linux standalone, it showed as "ruffle" (no extention), and opening it gave me a message about "this is a shared library file". I've tried anything I can think of- I've looked up the warning, tried using the gdebi app, tried opening it in Terminal and got a LibSSL error. I've looked for like any tutorial on this Ruffle version and found nothing about Linux, it's either just about Flash player or something. It amazes me that I have to even write about this prepostery, because it should be absolutely no problem OPENING a program FOR LINUX, as long as you're ON LINUX. Why would TAS Videos have instructions on Ruffle if that doesn't work, like do I need a different distro or something? I've seen nothing on that whatsoever on Ruffle's page. Can someone tell me anything about this?

**Comment by Lord-McSweeney** [comment, marked as answer]
> tried opening it in Terminal and got a LibSSL error

What version of Ubuntu are you on? Opening it in terminal is the right way to do it, but if you have anything older than Ubuntu 22.04, the prebuilt versions of Ruffle won't work, you'll have to build the project yourself.

**Comment by CurHob** [comment]
Ohhhh okay… So what you just said was confusing to me at first, because I had already updated and rebooted using the software updater application. But when I checked the version I had and saw it was like 20.something, then tried opening that again I realized there’s a difference between “update” (which I did), and “upgrade”. This version was installed a couple years ago.So I tried that, and I ran into problems restarting because I didn’t close Firefox as it asked (although why they would let you do that is mystifying in the first place), and had to boot advanced options and select dpkg. Thank goodness I can still boot Windows, but now that I’ve done that, I can finally open Ruffle and select it in libtas. If you can’t tell, this is still fairly new to me bc I haven’t used my Ubuntu partition very much, and it’s been a while since. Sorry to rant, it’s just kind of frustrating when there’s like nothing you can find to point to the solution, even and perhaps especially when it technically exists. Thanks though, I will now move on to attempt to get to know the basics of libtas.Sent from my iPhoneOn Apr 7, 2024, at 21:22, Lord-McSweeney ***@***.***> wrote:﻿

tried opening it in Terminal and got a LibSSL error

What version of Ubuntu are you on? Opening it in terminal is the right way to do it, but if you have anything older than Ubuntu 22.04, the prebuilt versions of Ruffle won't work, you'll have to build the project yourself.

**Comment by p0008874** [comment]
Please ask LibTAS-related questions in the LibTAS Discord server: https://discord.gg/3MBVAzU

**Comment by sombraguerrero** [comment]
@CurHob Also, if you don't want to go through the work of compiling yourself, I fairly regularly maintain a build for people to use here: https://sombrastorage.blob.core.windows.net/sombramain/ruffle_desktop

**Reply by torokati44** [reply]
Wait, why is this necessary? 😱 Why does the regular nightly release not work for you?

**Reply by Lord-McSweeney** [reply]
This is a known issue on Ubuntu 20, it has libssl 1.1 while the builds use libssl 3.

**Reply by sombraguerrero** [reply]
This is an old issue going almost two years back. We've had this in the Discord FAQ for a long time. Long story short is that libssl gets hard linked at compile time and hence, in both directions, the builds are incompatible with each other because newer Linux distro releases switched from shipping with libssl 1.1 to 3. This comes up in libTAS context a lot *because* libTAS generally gets run on older releases for the sake of less interference with X11 on account of Wayland.

**Reply by torokati44** [reply]
ah, okay, the "windows" in the URL confused me. 😵‍💫

**Reply by sombraguerrero** [reply]
Ah yes, I use an Azure storage container to host it. 😅
